src/main/kotlin/gg/ingot/iron/Iron.kt

```diff
@@ -104,10 +112,10 @@ class Iron(
      * Starts a transaction on the connection.
      * @since 1.0
      */
-    suspend fun <T : Any?> transaction(block: Controller.() -> T): T {
-        return withContext(dispatcher) { withController { controller ->
-            controller.transaction(block)
-        } }
+    suspend fun <T : Any?> transaction(block: Controller.() -> T): Result<T> {
```

Comment:
I'm not a fan of returning a result here, this could lead to unexpected and unwanted behaviour, what happens if the user who is using a transaction block is unaware of the return type, do we just not throw and allow them to continue the rest of the code causing issues? All queries and transactions should throw on failure, and for that failure to be handled in the calling method

Reply:
Turns out this change comes from #13 and not the master branch, make PRs unrelated to one another as much as possible, any merge conflicts can be resolved before being pushed fairly easily in these types of cases

Reply:
Hm, that's a completely fair point to make.
When implementing this change I was thinking for a value to always be returned however, users may use a transaction purely to do executions and not care about the return type.

How about making a new function expressly for the purpose of returning a Result on failure, or should we make the consumer make a wrapper over our transaction to do that themselves?

Using the Result type instead of blindly relying on thrown exceptions has been extremely crucial in writing organized code that is easy to follow and maintain, but I absolutely see what you mean by consumers who don't care about the return type.

Reply:
I'd prefer the user making a wrapper over the transaction, there could be things that require the db transaction to be successful before continuing the flow of things, for example in our use case we use our operator to move resources around to match our desired state in the database, if we forgot to .getOrThrow() in that scenario the operator would've either thrown itself later down the road or attempted to replicate an invalid state.

We could separate it into two methods but at that point it might get confusing which to use and when, and also the possibility of mistakenly using the wrong one could arise. I'd prefer throwing an exception for safer code  rather than slightly more organized code, but that's what I would prefer.

Reply:
I think this functionality in Iron itself would prove very beneficial for the overall DevEx, we should just properly document how the Result transaction works and go from there.

This isn't inherently a security vulnerability more-so people not understanding the tools they're using.